server/services/FiltersService/index.js

```diff
@@ -94,7 +95,7 @@ class FiltersService {
         {
           $match: {
             assessment: SOCIODEMOGRAPHICS_FORM,
-            study: { $in: this.userAccess, $nin: STUDIES_TO_OMIT },
+            study: { $in: this.filters.sites, $nin: STUDIES_TO_OMIT },
```

Comment:
it seems like if this service knows how to take the shape of filters from `label/value` object arrays and transform them to a flat array of strings for the queries then the rest of the app can use a consistent shape. There might be 1 other place (transforming the filters form the form input), but otherwise is that correct?

Reply:
qs has limit of 20 array items, in the test data my user has 22, transforming it in the UI is a risk for users with access to over 20+ sites, and this isn't the only place that requires a list of sites in a string array format, the SubjectModel a level higher up does as well

Reply:
for example, if we had another function here, something like `sitesFromFilters` and used that in the query then we might be able to reap the benefit of consistent object shapes.

```suggestion
            study: { $in: this.sitesFromFilters, $nin: STUDIES_TO_OMIT },
```

```js
class FiltersService {
  get sitesFromFilters() {
    return this.filters.sites.map(siteObj => siteObj.value)
  }
}
```

Reply:
> qs has limit of 20 array items, in the test data my user has 22, transforming it in the UI is a risk for users with access to over 20+ sites, because of what qs is doing under the hood

This seems like a limitation no matter what direction we go in related to the discussion we're having, right? Don't both directions use an array query parameter?

Reply:
> this isn't the only place that requires a list of sites in a string array format, the SubjectModel a level higher up does as well.

Doesn't the SubjectModel use the FiltersService though?

Reply:
The BarChartService in the controller requires a string array, I did consider instantiating the filters service in the controller, but then it would be passed down as a parameter to the subject model and that just feels weird to do.

Reply:
> > this isn't the only place that requires a list of sites in a string array format, the SubjectModel a level higher up does as well.
> 
> Doesn't the SubjectModel use the FiltersService though?

Yeah it is a limitation from the UI now when it sends queryparams